src/main/kotlin/de/itemis/mps/gradle/downloadJBR/Plugin.kt

```diff
@@ -5,12 +5,26 @@ import org.gradle.api.Plugin
 import org.gradle.api.Project
 import org.gradle.api.tasks.Copy
 import org.apache.tools.ant.taskdefs.condition.Os
+import org.gradle.api.file.DirectoryProperty
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.Optional
 import java.io.File
+import javax.inject.Inject
 
-open class DownloadJbrConfiguration {
-    lateinit var jbrVersion: String
-    var distributionType : String? = null
-    var downloadDir: File? = null
+open class DownloadJbrConfiguration @Inject constructor(of: ObjectFactory) {
+    @get:Input
+    val jbrVersion: Property<String>  = of.property(String::class.java)
```

Comment:
Same issue: this is an extension, I don't think its properties should be annotated with `@Input` and the like.